default.nix

```diff
@@ -117,44 +124,53 @@ with stdenv.lib; let
 in rec {
   mkNodeModules = { src, extraEnvVars ? {} }:
     let
+      localDeps = map dirOfLocal (builtins.filter isLocal (builtins.attrValues (lock.dependencies)));
       # filter out everything except package.json and package-lock.json if possible
       # allows to avoid rebuilding node_modules if these two files didn't change
       filteredSrc =
         let
           origSrc = if src ? _isLibCleanSourceWith then src.origSrc else src;
           getRelativePath = path: removePrefix (toString origSrc + "/") path;
-          usedPaths = [ "package.json" "package-lock.json" ];
+          usedPaths = [ "package.json" "package-lock.json" ]
+            ++ localDeps;
+          dirOfRec = x: if dirOf x == "." || dirOf x == "/" then [x] else ([x] ++ dirOfRec (dirOf x));
+          usedPathsRec = builtins.concatMap dirOfRec usedPaths;
           cleanedSource = cleanSourceWith {
             src = src;
-            filter = path: type: elem (getRelativePath path) usedPaths;
+            filter = path: type: any (x: elem x usedPathsRec) (dirOfRec (getRelativePath path));
           };
         in if canCleanSource src then cleanedSource else src;
 
-      packageJson = filteredSrc + "/package.json";
-      packageLockJson = filteredSrc + "/package-lock.json";
+      peerDependencies = map (localDep: mkNodeModules { inherit extraEnvVars; src = src + ("/" + localDep); }) localDeps;
+
+      packageJson = src + "/package.json";
+      packageLockJson = src + "/package-lock.json";
       info = fromJSON (readFile packageJson);
       lock = fromJSON (readFile packageLockJson);
     in stdenv.mkDerivation ({
-      name = "${info.name}-${info.version}-node-modules";
+      name = "${normalizeName info.name}-node-modules-${info.version}";
 
       buildInputs = [ _nodejs jq ];
 
+      src = filteredSrc;
+
       npmFlags = npmFlagsNpm;
-      buildCommand = ''
+
+      phases = [ "unpackPhase" "buildPhase" ];
+      buildPhase = ''
         # do not run the toplevel lifecycle scripts, we only do dependencies
         jq '.scripts={}' ${packageJson} > ./package.json
-        cp ${packageLockJson} ./package-lock.json
-
         echo 'building npm cache'
         chmod u+w ./package-lock.json
         NODE_PATH=${npmModules} node ${./mknpmcache.js} ${cacheInput "npm-cache-input.json" lock}
 
         echo 'building node_modules'
         npm $npmFlags ci
         patchShebangs ./node_modules/
+        ${concatMapStringsSep "\n" (dep: "cp -r ${dep}/node_modules/* node_modules") peerDependencies}
 
         mkdir $out
-        mv ./node_modules $out/
+        cp -r ./node_modules $out/
```

Comment:
Why?

Reply:
Because `node_modules` can contain read-only peer dependencies now, and in that case it can't be moved.